core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java

```diff
@@ -9,19 +8,72 @@
 import java.util.Optional;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
     @Resource
     private JdbcTemplate jdbcTemplate;
 
+    private static final String PHONES_AND_COLORS_QUERY = "SELECT phones.id, phones.brand, phones.model, phones.price, " +
+            "phones.imageUrl, colors.code, phone2color.colorId " +
+            "FROM phones " +
+            "JOIN phone2color ON phones.id=phone2color.phoneId " +
+            "JOIN colors ON phone2color.colorId=colors.id";
+    private static final String PHONES_AND_COLORS_QUERY_WITH_LIMIT_AND_OFFSET = PHONES_AND_COLORS_QUERY + " offset ? limit ?";
+    private static final String INSERT_INTO_PHONES2COLORS = "INSERT INTO phone2color VALUES (?, ?)";
+    private static final String DELETE_FROM_PHONES2COLORS = "DELETE FROM phone2color WHERE colorId = ? and phoneId = ?";
+    private static final String UPDATE_PHONE_PRICE = "UPDATE phones SET price = ?";
+    private static final String INSERT_NEW_PHONE = "INSERT INTO phones (brand, model, price, imageUrl) values (?, ?, ?, ?)";
+
     public Optional<Phone> get(final Long key) {
-        throw new UnsupportedOperationException("TODO");
+        return getPhoneListWithColors().stream()
```

Comment:
This is bad approach. What if your DB has thousands of products? You will have to query all of them and then convert all rows into Phone objects, just to get one